### PR TITLE
1263: replaced all stats in utils.cpp

### PR DESCRIPTION
**Pull request by Horki**
##### Checklist

- [ ] Language changes are updated in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [X] The new behaviour is covered by tests

@fbs Hi, replaced all `stats` in utils.cpp; please test

### CODE REVIEW

**Comment by Horki** [comment]
part of #1263

File: src/utils.cpp [hunk 1]
```diff
@@ -604,62 +604,46 @@ namespace, it will throw an exception
...
   }
 
-  if (fstat(self_fd, &self_stat))
+  if (target_path.string().length() >= 64)
   {
-    errmsg << "fstat(self_fd): " << strerror(errno);
-    goto error;
+    throw MountNSException(
+        "Failed to compare mount ns with PID " + std::to_string(pid) +
+        ". The error was Reading mountNS would overflow buffer.");
```
Comment (danobi) [inline src/utils.cpp:634]: Do we still need this check? Isn't the 64 length limit b/c of `char buf[64]` which is deleted now?

Comment (Horki) [reply src/utils.cpp:634]: @danobi Hi, just want to be backward-compatible, so the same exceptions are thrown

Comment (Horki) [reply src/utils.cpp]: @danobi Hi, I have removed the overflow exception

Comment (danobi) [reply src/utils.cpp]: > Hi, just want to be backward-compatible, so the same exceptions are thrown

I don't think anyone will mind if there are less errors than before :)

**Review by danobi — APPROVED** [review]

**Comment by danobi** [comment]
Thanks!

**Pull request merged**